util/ranger/detacher.go

```diff
@@ -195,15 +195,17 @@ func DetachIndexConditions(conditions []expression.Expression, cols []*expressio
 }
 
 func removeAccessConditions(conditions, accessConds []expression.Expression) []expression.Expression {
-	for i := len(conditions) - 1; i >= 0; i-- {
+	removed, len := 0, len(conditions)
```

Comment:
I think the original is easier to read.

Reply:
But it will change the `conditions` completely, for example, if you pass [a=1, b=2], and `accessConds` is [a =1], then `conditions` will be `[b=2, b=2]` after this call.